doc/proposals/ansible-operator-webhooks.md

```diff
@@ -0,0 +1,81 @@
+# Ansible Operator Webhooks
+
+> Status: Draft
+    are necessary to support defaulting, though the addition of the default field to the OpenAPI spec may make this one 
+## Background
+- [Conversion webhooks](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#webhook-conversion)
+    are necessary to support CRD versioning
+- [Mutating webhooks](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook)
+    are necessary to support defaulting, though the addition of the default field to the OpenAPI spec may make this one 
+    less important in the long-term.
+- [Validating webhooks](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) 
+    are necessary for higher level validation of incoming resources, for example SAR checks. There is an open community 
+    issue explicitly requesting this feature: https://github.com/operator-framework/operator-sdk/issues/1658 .
+
+Currently Ansible-based operators have no webhook support at all, a user would have to write their own webhook server 
+and deploy it alongside the operator to get that functionality.
+
+## Goals
+- Allow the creation and use of conversion/validating/mutating webhooks implemented with Ansible
+
+## Non-Goals
+- Handle xWebhookConfiguration objects or associated Services
+- Handle cert management
+
+## Solution
+Add a webhooks section to the `watches.yaml`, that allows the user to specify lists of validating/mutating/conversion 
+webhooks that should be created. 
+
+The user should only need to provide a playbook or a role, and the url path for the webhook. 
+
+The rest of the webhook configuration, ie the creation of the Service that points to the webhook server, the Webhook 
+Configuration resources that lay out the rules and point to the proper Service, and the management of the certificates 
+for the webhook server, will be the responsiblity of the user. 
+
+The webhook configuration will look roughly as follows, with a big asterisk on the conversion webhooks which need some
+further investigation.
+
+```yaml
+---
+- version: v1alpha1
+  group: apps.example.com
+  kind: MyApp
+  role: /opt/ansible/roles/myapp
+  webhooks:
```

Comment:
Do we want to add a "timeout" such that we can cancel these processes early if they take too long?

Reply:
Possibly, the webhook resource also allows you to set a timeout on the request so ideally we'd want to match that, though I'm not sure how we would (I'll check if timeout is sent in the request object)